optuna/trial.py

```diff
@@ -702,7 +702,7 @@ def _adjust_discrete_uniform_high(name, low, high, q):
 
     r = high - low
 
-    if math.fmod(r, q) != 0:
+    if (decimal.Decimal(str(r)) % decimal.Decimal(str(q))) != 0:
```

Comment:
This works fine but I recommended to avoid comparison between `decimal.Decimal` and numerical objects.

```python
>>> decimal.Decimal("0.0") == 0.0
True
>>> decimal.Decimal("0.1") == 0.1
False
```